taskgroup.go

```diff
@@ -42,70 +39,55 @@ func New(ef ErrorFunc) *Group { return &Group{onError: ef} }
 func (g *Group) Go(task Task) *Group {
 	g.wg.Add(1)
```

Comment:
personal style preference you may not agree with: wg.Add before init() feels weird to me, I tend to put any "init a zero value" call as early as possible, and wg.Adds right before the corresponding `go`.

A reverse engineered rationale would be that if g.init panicked somehow, there would be a phantom add. But really, it's just a personal style thing and seeing it structured otherwise made me stutter briefly in reading.

Reply:
So, prior to these changes it was actually semantically necessary, because otherwise someone could call Wait while init was in-flight and induce a panic. 

However, that might not be necessary anymore, let me look at it again.